### PR TITLE
DEV-1466 customize status text and url

### PR DESCRIPTION
**Pull request by asproul**
ticket: [DEV-1466]

## Summary
* If configuration file does not exist then, use environment variables. 
* Customize the status text and url posted to the PR through environment variables.
* Replace 'dimagi/required-labels' with 'compliance/required-labels'. Should this be configurable too?

## Python UnitTests
```
→ python3 -m unittest -v
test_it_source_config_from_environment (tests.test_config.TestGenerateConfig) ... ok
test_it_source_config_from_environment_with_missing_conf (tests.test_config.TestGenerateConfig) ... ok
test_it_source_config_from_given_config_file (tests.test_config.TestGenerateConfig) ... ok
test_all_fails (tests.test_label_assertion.LabelAssertionTests) ... ok
test_all_passes (tests.test_label_assertion.LabelAssertionTests) ... ok
test_banned_fails (tests.test_label_assertion.LabelAssertionTests) ... ok
test_banned_passes (tests.test_label_assertion.LabelAssertionTests) ... ok
test_pr_with_no_labels_no_requirements (tests.test_label_assertion.LabelAssertionTests) ... ok
test_required_any_fails (tests.test_label_assertion.LabelAssertionTests) ... ok
test_required_any_passes (tests.test_label_assertion.LabelAssertionTests) ... ok

----------------------------------------------------------------------
Ran 10 tests in 0.008s

OK
```

[DEV-1466]: https://liveramp.atlassian.net/browse/DEV-1466

### CODE REVIEW

File: tests/test_config.py [hunk 1]
```diff
@@ -6,19 +6,6 @@
 
 
 class TestGenerateConfig(unittest.TestCase):
-    @patch('config.ConfigParser')
```
Comment (julianvmodesto) [inline tests/test_config.py:9]: Is it possible to retain some version of these tests?

File: utils.py [hunk 2]
```diff
@@ -43,17 +43,17 @@ def post_status(self, status_json):
...
         else:
-            description = "Label requirements not satisfied."
+            description = status_text
         response_json = {
             "state": "success" if passes_label_requirements else "failure",
-            "target_url": "",
+            "target_url": status_target_url,
             "description": description,
-            "context": APP_NAME,
+            "context": "compliance/required-labels",
```
Comment (julianvmodesto) [inline utils.py]: Seems like we could retain `APP_NAME` here?

**Comment by julianvmodesto** [comment]
These are actually nits – overall this looks good

**Review by julianvmodesto — APPROVED** [review]

**Comment by julianvmodesto** [comment]
lgtm, feel free to address nits or not

File: tests/test_config.py (hunk 1 above)
Comment (asproul) [reply tests/test_config.py:9]: the last commit re-adds the tests and ensures they are passing

File: utils.py (hunk 2 above)
Comment (asproul) [reply utils.py]: updated name to `compliance/required-labels` rather than liveramp and kept the "APP_NAME" for the context.

**Comment by asproul** [comment]
Tests are passing after the most recent changes

```
± |2.3.1| ~/code/required-labels |DEV-1466 ✓| → python3 -m unittest -v
test_it_reads_default_config_file (tests.test_config.TestGenerateConfig) ... ok
test_it_reads_given_config_file_from_environment_var (tests.test_config.TestGenerateConfig) ... ok
test_it_source_config_from_environment (tests.test_config.TestGenerateConfig) ... ok
test_it_source_config_from_environment_with_missing_conf (tests.test_config.TestGenerateConfig) ... ok
test_it_source_config_from_given_config_file (tests.test_config.TestGenerateConfig) ... ok
test_all_fails (tests.test_label_assertion.LabelAssertionTests) ... ok
test_all_passes (tests.test_label_assertion.LabelAssertionTests) ... ok
test_banned_fails (tests.test_label_assertion.LabelAssertionTests) ... ok
test_banned_passes (tests.test_label_assertion.LabelAssertionTests) ... ok
test_pr_with_no_labels_no_requirements (tests.test_label_assertion.LabelAssertionTests) ... ok
test_required_any_fails (tests.test_label_assertion.LabelAssertionTests) ... ok
test_required_any_passes (tests.test_label_assertion.LabelAssertionTests) ... ok

----------------------------------------------------------------------
Ran 12 tests in 0.013s

OK
```

**Pull request merged**